test/api_container_wait_test.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/alibaba/pouch/test/environment"
+	"github.com/alibaba/pouch/test/request"
+
+	"github.com/go-check/check"
+)
+
+// APIContainerWaitSuite is the test suite for container wait API.
+type APIContainerWaitSuite struct{}
+
+func init() {
+	check.Suite(&APIContainerWaitSuite{})
+}
+
+// SetUpTest does common setup in the beginning of each test.
+func (suite *APIContainerWaitSuite) SetUpTest(c *check.C) {
+	SkipIfFalse(c, environment.IsLinux)
+	PullImage(c, busyboxImage)
+}
+
+// TestWaitOk tests waiting a stopped container return 200.
+func (suite *APIContainerWaitSuite) TestWaitOk(c *check.C) {
+	cname := "TestWaitOk"
+
+	CreateBusyboxContainerOk(c, cname)
+	StartContainerOk(c, cname)
+	StopContainerOk(c, cname)
+
+	resp, err := request.Post("/containers/" + cname + "/wait")
+	c.Assert(err, check.IsNil)
+	CheckRespStatus(c, resp, 200)
+
+	DelContainerForceOk(c, cname)
+}
+
+// TestRunningContainer tests waiting a running container to stop, then returns 200.
+func (suite *APIContainerWaitSuite) TestRunningContainer(c *check.C) {
+	cname := "TestRunningContainer"
+
+	CreateBusyboxContainerOk(c, cname)
+	StartContainerOk(c, cname)
+
+	var (
+		err  error
+		resp *http.Response
+	)
+
+	chWait := make(chan *http.Response)
```

Comment:
tip: if we uses the wait channel for notify, we can use empty struct in golang 😄

Reply:
Emmm, I think this channel is not only used for notify but also transferring data to the variable `resp ` in the next. So I think we can't use empty struct here. WDYT?